src/vmm/benches/queue.rs

```diff
@@ -0,0 +1,196 @@
+            // SAFETY: queue has 1 desc chanin
+            let desc = unsafe { queue.pop(&mem).unwrap_unchecked() };
+            let mut head = Some(desc);
+            while let Some(d) = head {
+                head = d.next_descriptor();
+            }
+        })
+    });
+
+    set_dtable_one_chain(&rxq, 2);
+    c.bench_function("next_descriptor_2", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            // SAFETY: queue has 1 desc chanin
+            let desc = unsafe { queue.pop(&mem).unwrap_unchecked() };
+            let mut head = Some(desc);
+            while let Some(d) = head {
+                head = d.next_descriptor();
+            }
+        })
+    });
+
+    set_dtable_one_chain(&rxq, 4);
+    c.bench_function("next_descriptor_4", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            // SAFETY: queue has 1 desc chanin
+            let desc = unsafe { queue.pop(&mem).unwrap_unchecked() };
+            let mut head = Some(desc);
+            while let Some(d) = head {
+                head = d.next_descriptor();
+            }
+        })
+    });
+
+    set_dtable_one_chain(&rxq, 16);
+    c.bench_function("next_descriptor_16", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            // SAFETY: queue has 1 desc chanin
+            let desc = unsafe { queue.pop(&mem).unwrap_unchecked() };
+            let mut head = Some(desc);
+            while let Some(d) = head {
+                head = d.next_descriptor();
+            }
+        })
+    });
+
+    // Queue pop
+
+    set_dtable_many_chains(&rxq, 1);
+    c.bench_function("queue_pop_1", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            while let Some(desc) = queue.pop(&mem) {
+                std::hint::black_box(desc);
+            }
+        })
+    });
+
+    set_dtable_many_chains(&rxq, 4);
+    c.bench_function("queue_pop_4", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            while let Some(desc) = queue.pop(&mem) {
+                std::hint::black_box(desc);
+            }
+        })
+    });
+
+    set_dtable_many_chains(&rxq, 16);
+    c.bench_function("queue_pop_16", |b| {
+        b.iter(|| {
+            queue.next_avail = Wrapping(0);
+            while let Some(desc) = queue.pop(&mem) {
+                std::hint::black_box(desc);
+            }
+        })
+    });
+
+    c.bench_function("queue_add_used_1", |b| {
+        b.iter(|| {
+            queue.num_added = Wrapping(0);
+            queue.next_used = Wrapping(0);
+            for i in 0_u16..1_u16 {
+                let index = std::hint::black_box(i);
+                let len = std::hint::black_box(i + 1);
+                // SAFETY: will never panic.
+                unsafe {
+                    queue
+                        .add_used(&mem, index as u16, len as u32)
+                        .unwrap_unchecked()
+                };
```

Comment:
Well, in this case, we can just ignore the return value. It will have the same effect.
```suggestion
                let _ =  queue.add_used(&mem, index as u16, len as u32);
```
same thing everywhere else, really.

Reply:
done